azblob/zc_credential_shared_key.go

```diff
@@ -164,7 +164,7 @@ func (f *SharedKeyCredential) buildCanonicalizedResource(u *url.URL) string {
 	// params is a map[string][]string; param name is key; params values is []string
 	params, err := url.ParseQuery(u.RawQuery) // Returns URL decoded values
 	if err != nil {
-		panic(err)
+		sanityCheckFailed("parsing query parameters must succeed, otherwise there might be serious problems in the SDK/generated code")
```

Comment:
`sanityCheckFailed` seems like a shallow alias to an underlying panic. Is that accurate?